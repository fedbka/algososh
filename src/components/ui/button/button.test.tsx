import { expect, describe, test, vi } from "vitest";
import { Button } from "./button";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Тестирование компонента 'Button'", () => {

  test("Компонент 'Button' без текста формируется без ошибок", () => {
    render(<Button />);
    const btn = screen.getByRole("button");
    expect(btn).toMatchSnapshot();
  });

  test("Компонент 'Button' c текстом формируется без ошибок", () => {
    render(<Button text="text" />);
    const btn = screen.getByRole("button");
    expect(btn).toMatchSnapshot();
  });

  test("Компонент 'Button' со свойством 'disabled' формируется без ошибок", () => {
    render(<Button disabled={true} />);
    const btn = screen.getByRole("button");
    expect(btn).toMatchSnapshot();
  });

  test("Компонент 'Button' с индикацией загрузки формируется без ошибок", () => {
    render(<Button isLoader={true} />);
    const btn = screen.getByRole("button");
    expect(btn).toMatchSnapshot();
  });

  test("Компонент 'Button' с вызовом функции по нажатию обрабатывает вызов корректно, ", () => {
    const clickHandler = vi.fn(() => 0);
    const { container } = render(<Button onClick={clickHandler} />);
    const btn = container.firstChild;
    fireEvent.click(btn as Element);
    expect(clickHandler).toHaveBeenCalledOnce();
  });

})
