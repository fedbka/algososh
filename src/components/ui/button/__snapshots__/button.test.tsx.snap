// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестирование компонента 'Button' > Компонент 'Button' c текстом формируется без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary _button_f92531 undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Тестирование компонента 'Button' > Компонент 'Button' без текста формируется без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary _button_f92531 undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Тестирование компонента 'Button' > Компонент 'Button' с вызовом функции по нажатию обрабатывает вызов корректно,  1`] = `
<button
  class="text text_type_button text_color_primary _button_f92531 undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Тестирование компонента 'Button' > Компонент 'Button' с индикацией загрузки формируется без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary _button_f92531 undefined _loader_f92531 "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="_loader_icon_f92531"
    src="/src/images/icons/loader.svg"
  />
</button>
`;

exports[`Тестирование компонента 'Button' > Компонент 'Button' со свойством 'disabled' формируется без ошибок 1`] = `
<button
  class="text text_type_button text_color_primary _button_f92531 undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
